IntegrationExample/AppDelegate.swift

```diff
@@ -9,27 +9,55 @@ import Foundation
 import MimiCoreKit
 import MimiUXKit
 import MimiSDK
+import Combine
 
-final class AppDelegate: NSObject, UIApplicationDelegate {
+final class AppDelegate: NSObject, UIApplicationDelegate, ObservableObject {
     
     private let firmwareController: FirmwareControlling = PartnerFirmwareController()
     private let headphoneConnectivityController = PartnerHeadphoneConnectivityController()
     private var audioProcessingController: PartnerAudioProcessingController!
     
+    private var cancellables = Set<AnyCancellable>()
+    
     // MARK: UIApplicationDelegate
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
         
         startMimiCore()
-        do {
-            let session = try activateMimiProcessing(techLevel: firmwareController.getTechLevel())
-            self.audioProcessingController = PartnerAudioProcessingController(session: session, firmwareController: firmwareController)
-        } catch {
-            fatalError("Failed to launch Mimi Processing:  \(error.localizedDescription)")
-        }
+        
+        // Observe headphone connectivity state changes
+        observeHeadphoneConnectivityState()
+        
         return true
     }
     
+    // Simulate a headphone connection, and as a result, activate the processing session
+    func simulateHeadphoneConnection(isConnected: Bool) {
+        headphoneConnectivityController.simulateHeadphoneConnection(isConnected: isConnected)
+    }
+    
+    private func observeHeadphoneConnectivityState() {
+        headphoneConnectivityController.state
+            .sink { state in
```

Comment:
we need to capture self weakly here.

Reply:
Since `AppDelegate` doesn't really get deallocated during the app lifecycle, is it really necessary?